source/common/router/config_impl.h

```diff
@@ -972,7 +972,9 @@ class RouteEntryImplBase : public RouteEntryAndRoute,
     // avoid possible circular reference. For example, the WeightedClusterEntry (derived from
     // DynamicRouteEntry) will be member of the RouteEntryImplBase, so the owner_ should be nullptr.
     const RouteConstSharedPtr owner_;
-    const std::string cluster_name_;
+    // The dynamic route entry will not be accessed cross thread and it is safe to mark this be
+    // mutable.
+    mutable std::string cluster_name_;
```

Comment:
nit: probably better to have a setter to improve maintainability or remove `mutable`.

Reply:
But I think may be you are right. May be the route entry that support dynamic refresh needn't to inherit from dynamic route entry. 🤔